auditor/src/domain/record.rs

```diff
@@ -18,34 +18,212 @@ use quickcheck;
+/// # meta.insert("features", vec!["ssd", "gpu"]);
+/// #
+/// let record = RecordAdd::new("123/456", meta, components, start_time)?;
+/// # Ok(())
+/// # }
+/// ```
 #[derive(Serialize, Deserialize, Clone, Debug)]
 pub struct RecordAdd {
+    /// Unique identifier of the record.
     pub record_id: ValidName,
+    /// Meta information, a collection of key value pairs in the form of `String` -> `Vec<String>`.
     pub meta: Option<ValidMeta>,
+    /// List of components that are accounted for.
     pub components: Vec<Component>,
+    /// Start time of the record.
     pub start_time: DateTime<Utc>,
+    /// Stop time of the record.
     pub stop_time: Option<DateTime<Utc>>,
 }
 
+/// [`RecordUpdate`] represents a single accountable unit that is used to set the `stop_time` of a
+/// [`Record`].
+///
+/// Initially, records are added to Auditor by pushing a [`RecordAdd`], where the `stop_time` field
+/// is optional. To later set the `stop_time` of the record, push a [`RecordUpdate`] with the same
+/// `record_id` to auditor.
+///
+/// Use the constructor to build a new record.
+///
+/// # Note
+/// All strings must not include the characters `/()"<>\{}`.
+///
+/// When created using the constructor,
+/// the record is already valid in terms of all checks that
+/// Auditor performs when receiving it.
+///
+/// Currently, only the `stop_time` can be updated.
+/// Setting other fields such as `meta` or `components` has no effect.
+///
+/// # Examples
+///
+/// Create a valid record
+///
+/// ```
+/// # use auditor::domain::{Component, RecordUpdate};
+/// # use std::collections::HashMap;
+/// use chrono::{DateTime, TimeZone, Utc};
+///
+/// # fn main() -> Result<(), anyhow::Error> {
+/// let stop_time: DateTime<Utc> = Utc.with_ymd_and_hms(2023, 1, 1, 12, 0, 0).unwrap();
+/// let record = RecordUpdate::new("123456", HashMap::new(), Vec::new(), stop_time)?;
+/// # Ok(())
+/// # }
+/// ```
+/// Create a record with an invalid id:
+///
+/// ```should_panic
+/// # use auditor::domain::{Component, RecordUpdate};
+/// # use chrono::{DateTime, TimeZone, Utc};
+/// # use std::collections::HashMap;
+/// #
+/// # fn main() -> Result<(), anyhow::Error> {
+/// # let stop_time: DateTime<Utc> = Utc.with_ymd_and_hms(2023, 1, 1, 0, 0, 0).unwrap();
+/// let record = RecordUpdate::new("123/456", HashMap::new(), Vec::new(), stop_time)?;
+/// # Ok(())
+/// # }
+/// ```
+
 #[derive(Serialize, Deserialize, Clone, Debug)]
 pub struct RecordUpdate {
+    /// Unique identifier of the record.
     pub record_id: ValidName,
+    /// Meta information, a collection of key value pairs in the form of `String` -> `Vec<String>`.
     pub meta: Option<ValidMeta>,
+    /// List of components that are accounted for.
     pub components: Vec<Component>,
+    /// Start time of the record.
     pub start_time: Option<DateTime<Utc>>,
+    /// Stop time of the record.
     pub stop_time: DateTime<Utc>,
 }
 
+/// A [`Record`] represents a single accountable unit.
```

Comment:
It's a bit strange to link to the `Record` docs, when you actually are currently looking at the `Record` docs. I think you may have done the same also in the code above.

Reply:
True, I will go through the rest and check that this is not done anywhere else.